buffer/src/test/java/io/netty/buffer/SlicedByteBufTest.java

```diff
@@ -224,4 +228,26 @@ public void testWriteIso88591CharSequenceExpand() {
     public void testWriteUtf16CharSequenceExpand() {
         super.testWriteUtf16CharSequenceExpand();
     }
+
+    @Test
+    public void ensureWritableWithEnoughSpaceShouldNotThrow() {
+        ByteBuf buf = buffer(1);
```

Comment:
I think you want to use `newBuffer(2)` to actually use a slice and also ensure we have enough space.